File: scripts/enhance-loop/measure.py
#!/usr/bin/env python3
"""Measure MCP tool latency and deps vulnerabilities."""
import json, time, subprocess, pathlib, os

def mcp_latency():
    try:
        from ccxt_gateway.mcp_server import SessionManager
        sm = SessionManager()
        sid = sm.start_session()
        t0 = time.perf_counter()
        sm.eval_in_session(sid, "1+1")
        t1 = time.perf_counter()
        sm.stop_session(sid)
        return t1 - t0
    except Exception as e:
        return None

def cache_size():
    d = pathlib.Path.home() / ".julia" / "compiled"
    if not d.exists():
        return 0
    return sum(f.stat().st_size for f in d.rglob("*") if f.is_file())

def main():
    metrics = {
        "timestamp": time.time(),
        "cache_bytes": cache_size(),
        "mcp_eval_s": mcp_latency(),
        "python": os.sys.version,
    }
    out = pathlib.Path(__file__).resolve().parents[2] / "reports" / "enhance-loop-baseline.json"
    out.parent.mkdir(parents=True, exist_ok=True)
    # merge with julia metrics if exists
    if out.exists():
        try:
            prev = json.loads(out.read_text())
            metrics = {**prev, **metrics}
        except: pass
    out.write_text(json.dumps(metrics, indent=2))
    print(json.dumps(metrics, indent=2))

if __name__ == "__main__":
    main()

File: scripts/enhance-loop/measure.jl
#!/usr/bin/env julia
"""Measure baseline metrics: Julia cold-start, precompile cache, deps."""
using Dates, JSON

function measure_cold_start(project::String="Planar")
    t0 = time()
    # We measure via subprocess to avoid current process warm cache
    t = @elapsed run(`julia --project=$project -e "using Planar"`)
    # Alternative: just time the subprocess externally; here we fallback to elapsed
    return t
end
function cache_size()
    dir = joinpath(homedir(), ".julia", "compiled")
    isdir(dir) || return 0
    total = 0
    for (root, _, files) in walkdir(dir)
        for f in files
            p = joinpath(root, f)
            try; total += filesize(p); catch; end
        end
    end
    return total
end
function main()
    metrics = Dict(
        "timestamp" => string(now(UTC)),
        "julia_version" => string(VERSION),
        "cache_bytes" => cache_size(),
        "cold_start_s" => try
            @elapsed run(`julia --project=Planar -e "using Planar; println(\"ok\")"`)
        catch e
            -1.0
        end,
    )
    out = joinpath(@__DIR__, "../../reports/enhance-loop-baseline.json")
    mkpath(dirname(out))
    open(out, "w") do io; JSON.print(io, metrics, 2); end
    println(JSON.json(metrics))
end
main()
